qt/widgets/common/src/FunctionModel.cpp

```diff
@@ -401,7 +409,9 @@ QString FunctionModel::getLocalParameterConstraint(const QString &parName,
 
 void FunctionModel::setLocalParameterValue(const QString &parName, int i,
                                            double value) {
-  getSingleFunction(i)->setParameter(parName.toStdString(), value);
+  auto function = getSingleFunction(i);
+  if (function && function->hasParameter(parName.toStdString()))
```

Comment:
There seems to be a small change in behavior I think?  It seems if we called `setParameter` with a typo, e.g `setParameter("Sigmaa", 1.00)` we'd fail silently now, whereas before we'd have got a `invalid_argument` error. 
Likewise if we called `setParameter` before initializing the function we'd have got a `logic_error("function is undefined")` now we'd get nothing.

Not entirely sure which situation was better, but was there a reason for this change?

Reply:
My assumption here was that each domain could have a different fit function. So when looping through each domain in `setGlobalParameterValue`, which calls `setLocalParameterValue` on each domain, I didn't want it to fail if one of the domains had a different function. I am not sure if the `FunctionModel` allows different functions for each domain, but I thought it would be preferable to do it this way?

Reply:
Ah I see, that makes sense. But wouldn't that situation where we have globals, but a different fit function defined at one domain index fail when we call `getFitFunctionWithGlobals`. As there we get each global parameter and construct a tie. E.g if A0 was a global parameter we essentially do 

```
for k = 1:nFunctions
    ties << fk.A0
ties.join("=").
```
which would yield `f0.A0=f1.A0=f2.A0 ect...` but the 2nd function may not have the A0 parameter.

Let me know if I've understood something wrong here. But, I wonder whether it's worthwhile worrying about this situation, I'm not sure it would make sense to have a global parameter that is not present in one of the functions. I just think its more likely that the hasParameter check will cover up a geniune code error, rather than this protecting against this edge case.

Reply:
Thats a good point, it should probably fail in that case. I'll change this back 👍